contracts/WitnetRequestsBoard.sol

```diff
@@ -239,6 +247,10 @@ contract WitnetRequestsBoard is WitnetRequestsBoardInterface {
     resultNotIncluded(_id)
     absMember(msg.sender)
  {
+    // Ensures the epoch when the dr inclusion was reported is smaller or equal
+    // the last epoch included in the block relay
+    uint256 witnetEpoch = blockRelay.getLastEpoch();
+    require(requests[_id].epoch <= witnetEpoch, "the current epoch is not compatible with the dr epoch");
```

Comment:
I think that rather than checking here if "the request was published in the past" we should check that "the result was published in a later block than the request".